modules/fx/arrow-fx-mtl/src/main/kotlin/arrow/fx/mtl/writert.kt

```diff
@@ -47,18 +47,17 @@ interface WriterTBracket<F, W> : Bracket<WriterTPartialOf<F, W>, Throwable>, Wri
     use: (A) -> WriterTOf<F, W, B>
   ): WriterT<F, W, B> = MM().run {
     MD().run {
-      WriterT(Ref(this, empty()).flatMap { ref ->
-        value().bracketCase(use = { wa ->
-          WriterT(wa.just()).flatMap(use).value()
-        }, release = { wa, exitCase ->
-          val r = release(wa.b, exitCase).value()
-          when (exitCase) {
-            is ExitCase.Completed -> r.flatMap { (l, _) -> ref.set(l) }
-            else -> r.unit()
-          }
-        }).flatMap { (w, b) ->
-          ref.get().map { ww -> Tuple2(w.combine(ww), b) }
+      val atomic: AtomicRefW<W> = AtomicRefW(empty())
```

Comment:
The `MonadDefer` constraint and the fix to `Throwable` can be removed now right? This means more flexible instances^^

Reply:
Just wondering, what's the reasoning to not use Ref here?

Reply:
Yes to both.
`release` is guaranteed to be called lazily after `use` and before it continues, so removing the wrapping `later` shouldn't be necessary to `defer`.

Reply:
I'm going to leave this for a separate ticket since this PR touches both Arrow Fx and Arrow Fx Mtl I'd rather close it now instead of re-targeting it after the re-org.